packages/react/src/Timeline/Timeline.tsx

```diff
@@ -68,6 +78,21 @@ const TimelineItem = styled.div.attrs<StyledTimelineItemProps>(props => ({
   ${sx};
 `
 
+export type TimelineItemsProps<As extends React.ElementType> = {
+  as?: As
+  condensed?: boolean
+} & React.ComponentPropsWithRef<React.ElementType extends As ? 'div' : As> &
+  SxProp
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function TimelineItem<As extends React.ElementType>(props: TimelineItemsProps<As>, ref: ForwardedRef<any>) {
```

Comment:
Taking a page out of [Branchname](https://github.com/primer/react/blob/main/packages/react/src/BranchName/BranchName.tsx#L31) here, open to suggestion